lib/intercept/lexer.cc

```diff
@@ -61,9 +61,7 @@ void lcc::intercept::Lexer::NextToken() {
     }
 
     /// Pop empty macro expansions off the expansion stack.
-    while (not macro_expansion_stack.empty())
-        if (macro_expansion_stack.back().done())
-            macro_expansion_stack.pop_back();
+    std::erase_if(macro_expansion_stack, [](MacroExpansion& x){ return x.done(); });
```

Comment:
```suggestion
    std::erase_if(macro_expansion_stack, &MacroExpansion::done);
```
Unless I’m missing something, this should work and is terser